### DISCUSSION TITLE
Persistent Ethernet Connectivity Issues on MinisForum AtomMan Intel Ultra 9 with Realtek r8126 Driver (10.014.00-NAPI) and Kernel 6.11

### DISCUSSION THREAD

**Discussion by sueknock** [linuxmint/discussions#627] [Questions]
I am still experiencing persistent Ethernet connectivity issues on my MinisForum AtomMan Intel Ultra 9, running Linux Mint Wilma version 22 with kernel 6.11.0-061100-generic. Both Ethernet ports (enp86s0 and enp87s0) consistently show a “cable unplugged” status, despite updating to the latest Realtek r8126 driver (version 10.014.00-NAPI).

System Information:
Kernel version: 6.11.0-061100-generic
Driver version: r8126 10.014.00-NAPI

Network Interfaces:
enp86s0: <NO-CARRIER,BROADCAST,MULTICAST,UP>
enp87s0: <NO-CARRIER,BROADCAST,MULTICAST,UP>
WiFi Interface: wlp88s0f0 is functioning normally.

Ethernet Controller Information:
56:00.0 Ethernet controller: Realtek Semiconductor Co., Ltd. Device 8126 (rev 01)
57:00.0 Ethernet controller: Realtek Semiconductor Co., Ltd. Device 8126 (rev 01)

dmesg logs related to ethernet: 
[    1.537901] r8126: loading out-of-tree module taints kernel.
[    1.537907] r8126: module verification failed: signature and/or required key missing - tainting kernel.

I had previously posted in this forum a month ago and was told that the upcoming Linux kernel update would likely resolve these issues. After updating from kernel 6.8 to 6.11, my Ethernet ports still show <NO-CARRIER>, and the Ethernet lights on the device are not lighting up!

Troubleshooting Steps Already Taken:
1. Updated the kernel from 6.8 to 6.11.
2. Repeatedly emailed RealTek customer support and installed their new Linux r8126 driver they emailed me (newer version compared to the one on their website): r8126 version 10.014.00-NAPI.
3. Verified the Ethernet cables and router ports are functional.
4. Checked and confirmed LAN is enabled in BIOS, and energy-saving features that could impact Ethernet are disabled.
5. Verified the driver was loaded without critical errors beyond module signing issues (which should not affect functionality as Secure Boot is disabled).

Am I the only one this is happening to? Unable to find any solutions on forums. I am trying to run this mini-PC as a server and really need ethernet to work, have 10G ethernet for the rest of my devices.

**Comment by sueknock** [comment]
@rcalixte, @mtwebster anyone have any ideas who I can contact to help me out with this? I have had this computer for 4 months and Linux Mint has unfortunately caused issue after issue. Updated everything I could think of but can never get the hardware to work as it should. No ethernet still. RealTek couldn't help any further after giving me an updated version.

**Reply by TalZiv** [reply]
Hi,
Can you please share the driver you got from Realtek?
